DYN-9036: Preserve bindings to switch from animation to opacity

zeusongit opened the pull request:
### Purpose

Re-enable the style, where fade animation on zoom out would switch to simpler visibility style if the workspace has more that 150 nodes to improve performance. this was broken when code only nodes were implemented.

This was part of the defer node PR, but I extracted this change since we may not merge that in 3.6.

Before:
![DynamoSandbox_Y6mTmsnikx](https://github.com/user-attachments/assets/5137095d-f39a-48cb-ab6d-916ad7b5f2a2)

After:
![DynamoSandbox_y2dACaP5WU](https://github.com/user-attachments/assets/34d89f72-4e9f-42dc-b5e6-08b1b1568c39)


### Declarations

Check these if you believe they are true

- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB
- [ ] This PR introduces new feature code involve network connecting and is tested with no-network mode.

### Release Notes

Preserve bindings to switch from animation to opacity

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

aparajit-pratap commented:
> ### Purpose
> Re-enable the style, where fade animation on zoom out would switch to simpler visibility style if the workspace has more that 150 nodes to improve performance. this was broken when code only nodes were implemented.
> 
> This was part of the defer node PR, but I extracted this change since we may not merge that in 3.6.
> 
> ### Declarations
> Check these if you believe they are true
> 
> * [ ]  Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
> * [ ]  The level of testing this PR includes is appropriate
> * [ ]  User facing strings, if any, are extracted into `*.resx` files
> * [ ]  Snapshot of UI changes, if any.
> * [ ]  Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
> * [ ]  This PR modifies some build requirements and the readme is updated
> * [ ]  This PR contains no files larger than 50 MB
> * [ ]  This PR introduces new feature code involve network connecting and is tested with no-network mode.
> 
> ### Release Notes
> Preserve bindings to switch from animation to opacity
> 
> ### Reviewers
> (FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)
> 
> (FILL ME IN, optional) Any additional notes to reviewers or testers.
> 
> ### FYIs
> (FILL ME IN, Optional) Names of anyone else you wish to be notified of

@zeusongit, can you post a gif to illustrate this change?

aparajit-pratap commented on src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:
```diff
@@ -194,6 +194,10 @@ internal Canvas expansionBay
         private static BoolToVisibilityConverter _booleanToVisibilityConverter = new BoolToVisibilityConverter();
         private static EmptyToVisibilityCollapsedConverter _emptyToVisibilityCollapsedConverter = new EmptyToVisibilityCollapsedConverter();
         private static ZoomToVisibilityCollapsedConverter _zoomToVisibilityCollapsedConverter = new ZoomToVisibilityCollapsedConverter();
+        private static IValueConverter _sZoomFadeControl = SharedDictionaryManager.DynamoModernDictionary["SZoomFadeControl"] as IValueConverter;
```
I don't believe having the `_` to name private members aligns with the original coding standards for Dynamo. I'm not sure when this new style started making its way into the code.

zeusongit commented:
> > ### Purpose
> > Re-enable the style, where fade animation on zoom out would switch to simpler visibility style if the workspace has more that 150 nodes to improve performance. this was broken when code only nodes were implemented.
> > This was part of the defer node PR, but I extracted this change since we may not merge that in 3.6.
> > ### Declarations
> > Check these if you believe they are true
> > 
> > * [ ]   Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
> > * [ ]   The level of testing this PR includes is appropriate
> > * [ ]   User facing strings, if any, are extracted into `*.resx` files
> > * [ ]   Snapshot of UI changes, if any.
> > * [ ]   Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
> > * [ ]   This PR modifies some build requirements and the readme is updated
> > * [ ]   This PR contains no files larger than 50 MB
> > * [ ]   This PR introduces new feature code involve network connecting and is tested with no-network mode.
> > 
> > ### Release Notes
> > Preserve bindings to switch from animation to opacity
> > ### Reviewers
> > (FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)
> > (FILL ME IN, optional) Any additional notes to reviewers or testers.
> > ### FYIs
> > (FILL ME IN, Optional) Names of anyone else you wish to be notified of
> 
> @zeusongit, can you post a gif to illustrate this change?

added

zeusongit commented on src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:
```diff
@@ -209,13 +213,14 @@ internal Canvas expansionBay
         private static Style _nodeButtonStyle = GetNodeButtonStyle();
         private static Style _codeBlockNodeItemControlStyle = GetCodeBlockPortItemControlStyle();
         internal static readonly Style DynamoToolTipTopStyle = GetDynamoToolTipTopStyle();
+        private static Binding _sZoomFadeControlStyleBinding = GetFadeToOpacityStyleBinding(_sZoomFadeControl);
+        private static Binding _sZoomFadeInControlStyleBinding = GetFadeToOpacityStyleBinding(_sZoomFadeInControl);
+        private static Binding _sZoomFadeInPreviewStyleBinding = GetFadeToOpacityStyleBinding(_sZoomFadeInPreview);
+        private static Binding _sZoomFadeOutPreviewStyleBinding = GetFadeToOpacityStyleBinding(_sZoomFadeOutPreview);
```
these were migrated from earlier xaml implementation as is

zeusongit commented on src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs:
```diff
@@ -1260,222 +1265,6 @@ private static Style GetDynamoToolTipTopStyle()
             return customTooltipStyle;
         }
 
-        private static Style GetZoomFadeOpacity_50PercentToZeroStyle()
```
Sure, it is roote from one of the Reope's PR: https://github.com/DynamoDS/Dynamo/pull/15749
It was observed that:
> WPF animations get pre-compiled to native platform code and that is why we can't get further details in both VS and dotTrace.
This is further exasperated by the fact that the animations target the contols' opacity property. Apparently when you modify an element’s opacity property, it can cause WPF to create temporary surfaces which results in an even bigger performance hit.

So basically animations in WPF are not performant.
